IMPORTING LIBRARIES

In [25]:
import pandas as pd

READING THE DATA

In [26]:
data = pd.read_csv("BCG TASK 1.csv")
data

Company  Year  Total Revenue   Net Income   Total Assets  \
0  Microsoft  2024   245122000000  88136000000   512163000000   
1  Microsoft  2023   211915000000   7236100000   411976000000   
2  Microsoft  2022   198270000000  72738000000   364840000000   
3      Tesla  2024    97690000000   7153000000   122070000000   
4      Tesla  2023    96773000000  14974000000   106618000000   
5      Tesla  2022    81462000000  12587000000    82338000000   
6      Apple  2024   391035000000  93736000000   364980000000   
7      Apple  2023   383285000000  96995000000   352583000000   
8      Apple  2022   394328000000  99803000000   352755000000   

   Total Liabilities      Cashflow  
0       243686000000  118548000000  
1       205753000000   87582000000  
2       198298000000   89035000000  
3        48390000000   14932000000  
4        43009000000   13256000000  
5        36440000000   14724000000  
6       308030000000  118254000000  
7       290437000000  110543000000  
8       302083000000  122151000000

In [27]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Company            9 non-null      object
 1   Year               9 non-null      int64 
 2   Total Revenue      9 non-null      int64 
 3   Net Income         9 non-null      int64 
 4    Total Assets      9 non-null      int64 
 5   Total Liabilities  9 non-null      int64 
 6   Cashflow           9 non-null      int64 
dtypes: int64(6), object(1)
memory usage: 636.0+ bytes


In [28]:
data.describe()

Year  Total Revenue    Net Income   Total Assets  \
count     9.000000   9.000000e+00  9.000000e+00   9.000000e+00   
mean   2023.000000   2.333200e+11  5.481757e+10   2.967026e+11   
std       0.866025   1.300070e+11  4.280094e+10   1.532404e+11   
min    2022.000000   8.146200e+10  7.153000e+09   8.233800e+10   
25%    2022.000000   9.769000e+10  1.258700e+10   1.220700e+11   
50%    2023.000000   2.119150e+11  7.273800e+10   3.527550e+11   
75%    2024.000000   3.832850e+11  9.373600e+10   3.649800e+11   
max    2024.000000   3.943280e+11  9.980300e+10   5.121630e+11   

       Total Liabilities      Cashflow  
count       9.000000e+00  9.000000e+00  
mean        1.862362e+11  7.655833e+10  
std         1.145062e+11  4.827133e+10  
min         3.644000e+10  1.325600e+10  
25%         4.839000e+10  1.493200e+10  
50%         2.057530e+11  8.903500e+10  
75%         2.904370e+11  1.182540e+11  
max         3.080300e+11  1.221510e+11

In [29]:
data.isna()

Company   Year  Total Revenue  Net Income   Total Assets  \
0    False  False          False       False          False   
1    False  False          False       False          False   
2    False  False          False       False          False   
3    False  False          False       False          False   
4    False  False          False       False          False   
5    False  False          False       False          False   
6    False  False          False       False          False   
7    False  False          False       False          False   
8    False  False          False       False          False   

   Total Liabilities  Cashflow  
0              False     False  
1              False     False  
2              False     False  
3              False     False  
4              False     False  
5              False     False  
6              False     False  
7              False     False  
8              False     False

In [30]:
data = data.sort_values(['Company','Year'],ascending=True).groupby('Company').head()
data

Company  Year  Total Revenue   Net Income   Total Assets  \
8      Apple  2022   394328000000  99803000000   352755000000   
7      Apple  2023   383285000000  96995000000   352583000000   
6      Apple  2024   391035000000  93736000000   364980000000   
2  Microsoft  2022   198270000000  72738000000   364840000000   
1  Microsoft  2023   211915000000   7236100000   411976000000   
0  Microsoft  2024   245122000000  88136000000   512163000000   
5      Tesla  2022    81462000000  12587000000    82338000000   
4      Tesla  2023    96773000000  14974000000   106618000000   
3      Tesla  2024    97690000000   7153000000   122070000000   

   Total Liabilities      Cashflow  
8       302083000000  122151000000  
7       290437000000  110543000000  
6       308030000000  118254000000  
2       198298000000   89035000000  
1       205753000000   87582000000  
0       243686000000  118548000000  
5        36440000000   14724000000  
4        43009000000   13256000000  
3        48390000000   14932000000

ANALYZING EACH COMPANY'S GROWTH

% OF REVENUE GROWTH

In [31]:
data_revenue = pd.DataFrame(columns=['Company','Year','Total Revenue','% of Revenue Growth'])
data_revenue['Company'] = data['Company']
data_revenue['Year'] = data['Year']
data_revenue['Total Revenue'] = data['Total Revenue']
data_revenue['% of Revenue Growth'] = data.groupby(['Company'])['Total Revenue'].pct_change() * 100
data_revenue

Company  Year  Total Revenue  % of Revenue Growth
8      Apple  2022   394328000000                  NaN
7      Apple  2023   383285000000            -2.800461
6      Apple  2024   391035000000             2.021994
2  Microsoft  2022   198270000000                  NaN
1  Microsoft  2023   211915000000             6.882030
0  Microsoft  2024   245122000000            15.669962
5      Tesla  2022    81462000000                  NaN
4      Tesla  2023    96773000000            18.795267
3      Tesla  2024    97690000000             0.947578

APPLE:  After a revenue dip in 2023 (-2.80% growth), Apple bounced back with 2.02% growth in 2024.
MICROSOFT:  Consistent growth each year, with a remarkable 15.67% revenue growth in 2024.
TESLA:  Strong growth in 2023 (18.79%), followed by minimal growth in 2024 (0.95%).
Microsoft showed the strongest upward trend in revenue growth over the three years, while Tesla's growth slowed in 2024. Apple rebounded after a decline in 2023. 



% OF NET INCOME GROWTH

In [32]:
data_income = pd.DataFrame(columns=['Company','Year','Net Income','% of Net Income Growth'])
data_income['Company'] = data['Company']
data_income['Year'] = data['Year']
data_income['Net Income'] = data['Net Income']
data_income['% of Net Income Growth'] = data.groupby(['Company'])['Net Income'].pct_change() * 100
data_income

Company  Year   Net Income  % of Net Income Growth
8      Apple  2022  99803000000                     NaN
7      Apple  2023  96995000000               -2.813543
6      Apple  2024  93736000000               -3.359967
2  Microsoft  2022  72738000000                     NaN
1  Microsoft  2023   7236100000              -90.051830
0  Microsoft  2024  88136000000             1118.004174
5      Tesla  2022  12587000000                     NaN
4      Tesla  2023  14974000000               18.964010
3      Tesla  2024   7153000000              -52.230533

APPLE: A decrease of 2.81% in 2023 followed by a 3.36% decrease in 2024 signals ongoing challenges.
MICROSOFT: A significant rebound in 2024, with net income growth at an impressive 1118.00% after a decline in 2023 (-90.05%).
TESLA: Positive net income growth in 2023 (18.96%), but a sharp decline of -52.23% in 2024 indicates potential volatility.
Microsoft shows strong recovery, while Tesla experiences a dramatic shift in 2024. Apple, meanwhile, faces consistent declines.

% OF ASSETS GROWTH

In [40]:
data_assets = pd.DataFrame(columns=['Company','Year','Total Assets','% of Assets Growth (%)'])
data_assets['Company'] = data['Company']
data_assets['Year'] = data['Year']
data_assets['Total Assets'] = data['Total Assets']
data_assets['% of Assets Growth'] = data.groupby(['Company'])['Total Assets'].pct_change() * 100
data_assets

Company  Year  Total Assets % of Assets Growth (%)  % of Assets Growth
8      Apple  2022  352755000000                    NaN                 NaN
7      Apple  2023  352583000000                    NaN           -0.048759
6      Apple  2024  364980000000                    NaN            3.516052
2  Microsoft  2022  364840000000                    NaN                 NaN
1  Microsoft  2023  411976000000                    NaN           12.919636
0  Microsoft  2024  512163000000                    NaN           24.318650
5      Tesla  2022   82338000000                    NaN                 NaN
4      Tesla  2023  106618000000                    NaN           29.488207
3      Tesla  2024  122070000000                    NaN           14.492862

In [38]:
print([col for col in data.columns])

['Company', 'Year', 'Total Revenue', 'Net Income', ' Total Assets', 'Total Liabilities', 'Cashflow']


In [39]:
data.columns = data.columns.str.strip()

APPLE: Despite a minor decline in assets in 2023 (-0.0488%), Apple rebounded with asset growth of 3.52% in 2024.
MICROSOFT: Strong and consistent growth, with 12.92% asset growth in 2023 and an impressive 24.32% in 2024.
TESLA: Exceptional asset growth in 2023 (29.49%), followed by a slower but still positive growth rate of 14.49% in 2024.
Microsoft exhibits the most robust and consistent upward trend, while Tesla's asset growth is slowing. Apple, though initially declining, recovers in 2024.

% OF LIABILITIES GROWTH

In [41]:
data_liability = pd.DataFrame(columns=['Company','Year','Total Liabilities','% of Liabilities Growth'])
data_liability['Company'] = data['Company']
data_liability['Year'] = data['Year']
data_liability['Total Liabilities'] = data['Total Liabilities']
data_liability['% of Liabilities Growth'] = data.groupby(['Company'])['Total Liabilities'].pct_change() * 100
data_liability

Company  Year  Total Liabilities  % of Liabilities Growth
8      Apple  2022       302083000000                      NaN
7      Apple  2023       290437000000                -3.855232
6      Apple  2024       308030000000                 6.057424
2  Microsoft  2022       198298000000                      NaN
1  Microsoft  2023       205753000000                 3.759493
0  Microsoft  2024       243686000000                18.436183
5      Tesla  2022        36440000000                      NaN
4      Tesla  2023        43009000000                18.026894
3      Tesla  2024        48390000000                12.511335

APPLE: Liabilities declined in 2023 (-3.86%) but grew by 6.06% in 2024.
MICROSOFT: Consistent growth in liabilities, with a modest increase of 3.76% in 2023, followed by significant growth of 18.44% in 2024.
TESLA: Strong liabilities growth in 2023 (18.03%) and continued but slower growth in 2024 (12.51%).
Microsoft leads in liabilities growth for 2024, while Tesla demonstrates consistent but tapering increases. Apple shows a recovery trend after a decrease.

% OF CASHFLOW

In [46]:
data_cashflow = pd.DataFrame(columns=['Company','Year','Cashflow','% of Cashflow Growth'])
data_cashflow['Company'] = data['Company']
data_cashflow['Year'] = data['Year']
data_cashflow['Cashflow'] = data['Cashflow']
data_cashflow['% of Cashflow Growth'] = data.groupby(['Company'])['Cashflow'].pct_change() * 100
data_cashflow

Company  Year      Cashflow  % of Cashflow Growth
8      Apple  2022  122151000000                   NaN
7      Apple  2023  110543000000             -9.502992
6      Apple  2024  118254000000              6.975566
2  Microsoft  2022   89035000000                   NaN
1  Microsoft  2023   87582000000             -1.631942
0  Microsoft  2024  118548000000             35.356580
5      Tesla  2022   14724000000                   NaN
4      Tesla  2023   13256000000             -9.970117
3      Tesla  2024   14932000000             12.643331

APPLE: After a significant decline in 2023 (-9.50%), Apple recovered with a 6.98% increase in cash flow for 2024.
MICROSOFT: Minor decline in 2023 (-1.63%), followed by an impressive growth of 35.36% in 2024.
TESLA: Sharp drop in 2023 (-9.97%), but a strong rebound in 2024 with a 12.64% increase in cash flow.
Microsoft demonstrated the most remarkable recovery and growth in 2024, while Tesla and Apple showed positive trends after declines in 2023.